docs/resources.md

```diff
@@ -0,0 +1,91 @@
+# Images, fonts and other resources
+
+Typst packages can ship more than just code. This document explains how to
+properly handle resource files and avoid common mistakes.
+
+## Paths are package scoped
+
+When reading external files, the path that is given to the function
+(like `image`, `json` or `read`) is relative to the current file
+and can only access file in the current package. This means it can't
+access files from other packages. But more importantly, in case your package
+is a template, it can't access files in the users project.
+
+For instance, let's say you have the following file hierarchy:
+
+```
+preview/my-template/1.0.0/
+  README.md
+  LICENSE
+  typst.toml
+  thumbnail.png
+  src/
+    lib.typ
+  template/
+    main.typ
+    logo.png
+```
+
+In `lib.typ`, calling `image("../template/logo.png")` will seem to work.
+But this won't refer to the copy of `logo.png` in the users project: if they
+were to replace the provided logo with their own, your package would still use
+the original one.
+
+It also means that allowing for customization using a string parameter called `logo-path`
+that would then give to `image` in your package will not work either: the file
+access is made relatively to where the `image` function is called, not to where
+the path is created.
+
+The proper way to let the people using your template overwrite the file to use
+is to take `content` as an argument directly, not a `string`. For example, you should
+replace this:
+
+```typ
+#let cover-page(logo-path: "logo.png", title) = {
+  image(logo-path)
+  heading(title)
+}
+```
+
+With something like:
+
+```typ
+#let cover-page(logo: image("logo.png"), title) = {
+  logo
+  heading(title)
+}
+```
+
+Note that this means that the user will be able to provide arbitrary content and
+not necessarily an image.
```

Comment:
Not sure this is relevant enough to include. What I think would be more useful to point out is that you can use a set rule to still customize e.g. the width of the image.